packages/flutter/lib/src/material/scrollbar.dart

```diff
@@ -2,10 +2,186 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:async';
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
 import 'package:flutter/widgets.dart';
 
 import 'theme.dart';
 
+class Scrollbar2 extends StatefulWidget {
+  Scrollbar2({
+    Key key,
+    this.child,
+  }) : super(key: key);
+
+
+  /// The subtree to place inside the [Scrollbar2]. This should include
+  /// a source of [ScrollNotification2] notifications, typically a [Scrollable2]
+  /// widget.
+  final Widget child;
+
+  @override
+  _Scrollbar2State createState() => new _Scrollbar2State();
+}
+
+class _Scrollbar2State extends State<Scrollbar2> with TickerProviderStateMixin {
+  _ScrollbarController _controller;
+
+  @override
+  void dependenciesChanged() {
+    super.dependenciesChanged();
+    _controller ??= new _ScrollbarController(this);
+    _controller.color = Theme.of(context).highlightColor;
+  }
+
+  bool _handleScrollNotification(ScrollNotification2 notification) {
+    if (notification is ScrollUpdateNotification ||
+        notification is OverscrollNotification)
+      _controller.update(notification.metrics, notification.axisDirection);
+    return false;
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new NotificationListener<ScrollNotification2>(
+      onNotification: _handleScrollNotification,
+      child: new RepaintBoundary(
```

Comment:
is there some way we could make CustomPaint take a flag that decides whether to do it or not?